packages/core/src/tracing/instrumentation/messaging/rdkafka.js

```diff
@@ -138,20 +138,22 @@ function instrumentedProduce(ctx, originalProduce, originalArgs) {
     typeof ctx._cb_configs.event.delivery_cb === 'function';
 
   return cls.ns.runAndReturn(() => {
+    const topic = originalArgs[0];
+    const spanData = {
+      kafka: {
+        endpoints: topic,
+        operation: 'send'
+      }
+    };
     const span = cls.startSpan({
       spanName: 'kafka',
-      kind: constants.EXIT
+      kind: constants.EXIT,
+      spanData
     });
-    const topic = originalArgs[0];
 
     span.stack = tracingUtil.getStackTrace(instrumentedProduce, 1);
-    span.data.kafka = {
-      service: topic,
-      access: 'send'
-    };
 
-    const headers = addTraceContextHeader(originalArgs[6], span);
-    originalArgs[6] = headers;
+    originalArgs[6] = setTraceHeaders(originalArgs[6], span);
```

Comment:
consistency request: in kafkajs instr the fn is written a bit differently

> setTraceHeaders({... isIgnored: ...})

Choose one version.